examples/lighting-app/qpg/src/AppTask.cpp

```diff
@@ -243,21 +223,52 @@ CHIP_ERROR AppTask::Init()
     initParams.endpointNativeParams    = static_cast<void *>(&nativeParams);
     chip::Server::GetInstance().Init(initParams);
 
-    // Init OTA engine
-    InitializeOTARequestor();
+#if CHIP_DEVICE_CONFIG_ENABLE_EXTENDED_DISCOVERY
+    chip::app::DnssdServer::Instance().SetExtendedDiscoveryTimeoutSecs(extDiscTimeoutSecs);
+#endif
+
+    // Open commissioning after boot if no fabric was available
+    if (chip::Server::GetInstance().GetFabricTable().FabricCount() == 0)
+    {
+        PlatformMgr().ScheduleWork(OpenCommissioning, 0);
+    }
+}
+
+void AppTask::OpenCommissioning(intptr_t arg)
+{
+    // Enable BLE advertisements
+    chip::Server::GetInstance().GetCommissioningWindowManager().OpenBasicCommissioningWindow();
+    ChipLogProgress(NotSpecified, "BLE advertising started. Waiting for Pairing.");
+}
+
+CHIP_ERROR AppTask::Init()
+{
+    CHIP_ERROR err = CHIP_NO_ERROR;
+
+    ChipLogProgress(NotSpecified, "Current Software Version: %s", CHIP_DEVICE_CONFIG_DEVICE_SOFTWARE_VERSION_STRING);
+
+    // Init ZCL Data Model and start server
+    PlatformMgr().ScheduleWork(InitServer, 0);
 
     // Initialize device attestation config
     SetDeviceAttestationCredentialsProvider(Examples::GetExampleDACProvider());
 
-    UpdateClusterState();
+    // Setup light
+    err = LightingMgr().Init();
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(NotSpecified, "LightingMgr().Init() failed");
+        return err;
+    }
+    LightingMgr().SetCallbacks(ActionInitiated, ActionCompleted);
 
+    // Setup button handler
+    qvIO_SetBtnCallback(ButtonEventHandler);
+
+    // Log device configuration
     ConfigurationMgr().LogDeviceConfig();
```

Comment:
Shouldn't this be called from the Matter threading context as well?

Reply:
We could indeed defer this as well, but the ConfigurationMgr() should be initialized at this stage and has no dependencies on the Server instance being initialized (I'm assuming).
Our underlying NVM is protected for multi-task access if this and any Matter task would be accessing this.
Is there an overview available what API's should/can be used in which context ?